### PR TITLE
feat: backport pipeline from #2260 [7.2]

### PR DESCRIPTION
**Pull request by v1v**
backport pipeline from https://github.com/elastic/apm-server/issues/2260

### CODE REVIEW

File: .ci/check-changelogs.groovy [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  triggers {
+    cron 'H H(3-4) * * 1-5'
+    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
```
Comment (kuisathaverat) [inline .ci/check-changelogs.groovy]: Should we change the sentence?

File: .ci/check-changelogs.groovy [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+  stages {
+    /**
+     Checkout the code and stash it, to use it on other stages.
+    */
+    stage('Checkout') {
+      agent { label 'master || immutable' }
+      environment {
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+        HOME = "${env.WORKSPACE}"
```
Comment (kuisathaverat) [inline .ci/check-changelogs.groovy]: Not needed
```suggestion
```

File: .ci/check-changelogs.groovy (hunk 1 above)
Comment (v1v) [reply .ci/check-changelogs.groovy]: That's a leftover, so I don't think we need them as the pipeline runs on a daily basis for the main branches.

File: .ci/check-changelogs.groovy [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        GOPATH = "${env.WORKSPACE}"
```
Comment (kuisathaverat) [inline .ci/check-changelogs.groovy]: ditto
```suggestion
```

File: .ci/check-changelogs.groovy [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      }
+      steps {
+        deleteDir()
+        unstash 'source'
+        dir("${BASE_DIR}"){
+          sh(label: 'Run check changelogs', script: './script/jenkins/check-changelogs.sh')
```
Comment (kuisathaverat) [inline .ci/check-changelogs.groovy]: Do it need Go? if so workers do not have it installed, we start using a docker Go container for isolation

**Comment by v1v** [comment]
I'll apply your comments in the remaining PRs and also add the remaining files :)

File: .ci/check-changelogs.groovy (hunk 4 above)
Comment (v1v) [reply .ci/check-changelogs.groovy]: not really, it's a dockerised script

**Comment by v1v** [comment]
After revisiting this PR I realize this is not required though as we the script itself should be branch independent as the list of branches are defined within the script. Therefore I'll close this PR now

**Pull request closed without merging**